day21/helpingStephan.py

```diff
@@ -0,0 +1,40 @@
+def helpingStephan(x, y):
+    xSplit = str(x).split(".")
+    ySplit = str(y).split(".")
+    
+    XdigitsAfterDecimal = "0"
+    if len(xSplit) > 1:
+        XdigitsAfterDecimal = xSplit[1]
+        
+    YdigitsAfterDecimal = "0"
+    if len(ySplit) > 1:
+        YdigitsAfterDecimal = ySplit[1]
+                                      
+                                  
+    digitsAfterDecimal = max(len(XdigitsAfterDecimal), len(YdigitsAfterDecimal))
+                
+    x *= 10**digitsAfterDecimal
+    y *= 10**digitsAfterDecimal
+
+    a = max(x, y)
+    b = min(x, y)
+     
+    r = -1
+    while r != 0:
+        r = round(a % b, 6) * 1.0
```

Comment:
do you need to do floating point math here? since you multiplied `x` and `y` by a power of 10 to make them effectively integers, can't you just run gcd in its integer form?